Submitted By: Alexander E. Patrakov
Date: 2004-05-09
Initial Package Version: 1.4.1
Origin: Debian
Upstream status: present in Debian
Description: Fixes the signal handling logic.
Without this patch, the following command can fail:

/etc/rc.d/init.d/sysklogd restart

diff -u sysklogd-1.4.1/syslogd.c sysklogd-1.4.1.signal/syslogd.c
--- sysklogd-1.4.1/syslogd.c	2001-03-12 00:40:10.000000000 +0500
+++ sysklogd-1.4.1.signal/syslogd.c	2004-05-09 10:17:28.603308784 +0600
@@ -890,11 +890,11 @@
 		dprintf("Checking pidfile.\n");
 		if (!check_pid(PidFile))
 		{
+			signal(SIGTERM, doexit);
 			if (fork()) {
 				/*
 				 * Parent process
 				 */
-				signal (SIGTERM, doexit);
 				sleep(300);
 				/*
 				 * Not reached unless something major went wrong.  5
